orchestrator/gravity_utils/src/types/config.rs

```diff
@@ -0,0 +1,24 @@
+use serde_derive::Deserialize;
+use std::str::FromStr;
+
+/// The various possible modes for relaying
+#[derive(Debug, Deserialize, PartialEq, Copy, Clone)]
+pub enum RelayerMode {
+    /// Always relay batches, profitable or not
+    AlwaysRelay,
+    /// Use private API to fetch the price data feed for the cost estimation
+    Api,
+    /// Use file to fetch the token price for the cost estimation
+    File,
+}
+
+impl FromStr for RelayerMode {
+    type Err = ();
+    fn from_str(input: &str) -> Result<RelayerMode, Self::Err> {
+        let mode = RelayerMode::from_str(input).map_err(|_|{
```

Comment:
even though the code seems to work I have a warning

```Cannot return without recursing```

Any suggestion @linfeng-crypto ?

Reply:
It does look like an infinite recursion? or if there's a different `from_str` on `RelayerMode` than from the `FromStr` trait implementation, then it can be a qualified call

Reply:
e.g. `let mode: RelayerMode = serde_json::from_str(input)...`

Reply:
not sure, I made the change due to @linfeng-crypto comment to use https://docs.rs/strum/latest/strum/derive.EnumString.html 
it uses auto derive?

Reply:
https://github.com/crypto-org-chain/gravity-bridge/pull/36/files#diff-7bb0813962d73124fc0dff4b50433282e43fdfa3e97adec12f581abab67d216eR5

own L5, there's no EnumString -- if you used EnumString, then there's no need to write FromStr implementation (it'll be autogenerated)

Reply:
if I remove it, I have the following error at compilation

```
error[E0599]: no variant or associated item named `from_str` found for enum `RelayerMode` in the current scope
   --> gorc/src/commands/orchestrator/start.rs:109:37
    |
109 |             let mode = RelayerMode::from_str(mode_str).expect("Incorrect mode, possible value are: AlwaysRelay, Api or File");
    |                                     ^^^^^^^^ variant or associated item not found in `RelayerMode`
   
```

Reply:
okay I have updated to use strum and it works. thanks 🙏